src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIClientService.java

```diff
@@ -1,48 +1,136 @@
  * UI Service to handle IRIDA Clients
  */
 @Component
 public class UIClientService {
-	private final IridaClientDetailsService clientDetailsService;
-
-	public UIClientService(IridaClientDetailsService clientDetailsService) {
-		this.clientDetailsService = clientDetailsService;
-	}
-
-	/**
-	 * Get a listing of clients based on the table request.
-	 *
-	 * @param tableRequest Information about the sort and page of the table.
-	 * @return Current status of the table
-	 */
-	public TableResponse<ClientTableModel> getClientList(ClientTableRequest tableRequest) {
-		Specification<IridaClientDetails> specification = IridaClientDetailsSpecification.searchClient(
-				tableRequest.getSearch());
-
-		Page<IridaClientDetails> page = clientDetailsService.search(specification,
-				PageRequest.of(tableRequest.getCurrent(), tableRequest.getPageSize(), tableRequest.getSort()));
-		List<ClientTableModel> models = page.getContent()
-				.stream()
-				.map(client -> new ClientTableModel(client, clientDetailsService.countActiveTokensForClient(client)))
-				.collect(Collectors.toList());
-
-		return new TableResponse<>(models, page.getTotalElements());
-	}
+    private final IridaClientDetailsService clientDetailsService;
+
+    public UIClientService(IridaClientDetailsService clientDetailsService) {
+        this.clientDetailsService = clientDetailsService;
+    }
+
+    /**
+     * Get a listing of clients based on the table request.
+     *
+     * @param tableRequest Information about the sort and page of the table.
+     * @return Current status of the table
+     */
+    public TableResponse<ClientTableModel> getClientList(ClientTableRequest tableRequest) {
+        Specification<IridaClientDetails> specification = IridaClientDetailsSpecification.searchClient(
+                tableRequest.getSearch());
+
+        Page<IridaClientDetails> page = clientDetailsService.search(specification,
+                PageRequest.of(tableRequest.getCurrent(), tableRequest.getPageSize(), tableRequest.getSort()));
+        List<ClientTableModel> models = page.getContent()
+                .stream()
+                .map(client -> new ClientTableModel(client, clientDetailsService.countActiveTokensForClient(client)))
+                .collect(Collectors.toList());
+
+        return new TableResponse<>(models, page.getTotalElements());
+    }
+
+    /**
+     * Revoke all tokens for a specific client
+     *
+     * @param id Identifier for a client
+     */
+    public void deleteClientTokens(Long id) {
+        IridaClientDetails details = clientDetailsService.read(id);
+        clientDetailsService.revokeTokensForClient(details);
+    }
+
+    /**
+     * Validate a client identifier for a new client
+     *
+     * @param clientId Identifier to check to see if it exists
+     * @throws ClientRegistrationException thrown if a client does not exist with the given client id.
+     */
+    public void validateClientId(String clientId) throws ClientRegistrationException {
```

Comment:
As noted above. If it is updated to throw a NoSuchClientException then this method signature will also need to be updated

Reply:
Updated in ecc4624